Services/Cron/classes/class.ilCronStartUp.php

```diff
@@ -27,35 +27,28 @@ public function __construct(
 		$a_client_id,
 		$a_login,
 		$a_password,
-		\ilAuthSession $authSession = null
+		ilAuthSession $authSession = null
 	) {
 		$this->client = $a_client_id;
 		$this->username = $a_login;
 		$this->password = $a_password;
 
-		if ($authSession) {
-			global $DIC;
-			$authSession = $DIC['ilAuthSession'];
-		}
-		$this->authSession = $authSession;
-	}
-	
-	/** 
-	 * Init ILIAS
-	 */
-	public function initIlias()
-	{
 		include_once './Services/Context/classes/class.ilContext.php';
 		ilContext::init(ilContext::CONTEXT_CRON);
-		
+
 		// define client
 		// @see mantis 20371
 		$_GET['client_id'] = $this->client;
-		
+
 		include_once './include/inc.header.php';
+
+		if (null === $authSession) {
+			global $DIC;
+			$authSession = $DIC['ilAuthSession'];
+		}
```

Comment:
Just for curiosity: Why is this situp required? Is there anything that prevents injecting the DIC at creation side?

Reply:
The instantiation of `ilCronStartUp` is at the very start of creation, even before the initialising of ILIAS.
AFAIK the DIC will created somewhere along `./include/inc.header.php` which means that I need the above code(see line 37 & 41).

I would love to create just an instance of `ilAuthSession` from scratch e.g.: `$authSession = new ilAuthSession($clientId, $logger)`, but currently this is not possible, that is why I currently need this weird setup.

I hope this answers your question :+1: